numpy/core/src/_simd/_simd_inc_easyintrin.h

```diff
@@ -0,0 +1,197 @@
+#ifndef NPY_SIMD
```

Comment:
These macros seem rather difficult. Could we use the repeat mechanism to generate this differently? It would be nice to avoid the `SIMD__IMPL_COUNT_` ones especially.

Reply:
> Could we use the repeat mechanism to generate this differently?

Do you mean using source templates? if yes, then no since that will increase the compiling time to over x2, also it will lead non-robust code.

>  It would be nice to avoid the SIMD__IMPL_COUNT_ ones especially

No way to avoid it, this marco is used for converting a runtime scalar(python integer) to an immediate constant.
But what about re-implement this module in C++/templates? Definitely it will increase the readability.

Reply:
heh, given the choice between these macros and C++ templates, I think I prefer the macros